crates/ide-assists/src/handlers/bool_to_enum.rs

```diff
@@ -93,7 +93,29 @@ struct BoolNodeData {
 fn find_bool_node(ctx: &AssistContext<'_>) -> Option<BoolNodeData> {
     let name: ast::Name = ctx.find_node_at_offset()?;
 
-    if let Some(let_stmt) = name.syntax().ancestors().find_map(ast::LetStmt::cast) {
+    if let Some(param) = name.syntax().ancestors().find_map(ast::Param::cast) {
```

Comment:
This can be simplified a bit (that is we can mostly collapse this and the let statement branch) by first looking for the ident pat, then branching on the parents of that instead (since both let statements and param positions take patterns)
```rs
if let Some(param) = name.syntax().ancestors().find_map(ast::IdentPat::cast) {
    .. do type checks and stuff as before
    match param.syntax().ancestors().find_map(Either::<ast::Param, ast::LetStmt>::cast)? {
        Either::Left(param) => populate BoolNodeData accordingly
        Either::Right(let_stmt=> populate BoolNodeData accordingly
    }
```

Reply:
I didn't know about `Either`, thank you!